crates/bevy_ecs/hecs/src/archetype.rs

```diff
@@ -218,11 +239,12 @@ impl Archetype {
     /// `index` must be in-bounds
     pub(crate) unsafe fn get_dynamic(
         &self,
-        ty: TypeId,
+        ty: ComponentId,
         size: usize,
         index: usize,
     ) -> Option<NonNull<u8>> {
-        debug_assert!(index < self.len);
+        // TODO(zicklag): I'm pretty sure that it is valid for the index to be zero
+        debug_assert!(index < self.len || index == 0);
```

Comment:
I'm pretty sure this insertion without the extra `|| index == 0` is flawed, as it isn't even upheld in the normal, non-dynamic `get` function. After accepting an index of 0 it apears to work fine.